statsd/telemetry.go

```diff
@@ -97,47 +192,85 @@ func (t *telemetryClient) sendTelemetry() {
 	}
 }
 
+func (t *telemetryClient) getTelemetry() Telemetry {
+	if t == nil {
+		// telemetry was disabled through the WithoutTelemetry option
+		return Telemetry{}
+	}
+
+	tlm := Telemetry{}
+	t.c.flushTelemetryMetrics(&tlm)
+	t.c.sender.flushTelemetryMetrics(&tlm)
+	t.c.agg.flushTelemetryMetrics(&tlm)
+
+	tlm.TotalMetrics = tlm.TotalMetricsGauge +
+		tlm.TotalMetricsCount +
+		tlm.TotalMetricsSet +
+		tlm.TotalMetricsHistogram +
+		tlm.TotalMetricsDistribution +
+		tlm.TotalMetricsTiming
+
+	tlm.TotalPayloadsDropped = tlm.TotalPayloadsDroppedQueueFull + tlm.TotalPayloadsDroppedWriter
+	tlm.TotalBytesDropped = tlm.TotalBytesDroppedQueueFull + tlm.TotalBytesDroppedWriter
+
+	if tlm.aggEnabled {
+		tlm.AggregationNbContext = tlm.AggregationNbContextGauge +
+			tlm.AggregationNbContextCount +
+			tlm.AggregationNbContextSet +
+			tlm.AggregationNbContextHistogram +
+			tlm.AggregationNbContextDistribution +
+			tlm.AggregationNbContextTiming
+	}
+	return tlm
+}
+
 // flushTelemetry returns Telemetry metrics to be flushed. It's its own function to ease testing.
 func (t *telemetryClient) flush() []metric {
 	m := []metric{}
 
 	// same as Count but without global namespace
-	telemetryCount := func(name string, value int64, tags []string) {
+	telemetryGauge := func(name string, value int64, tags []string) {
```

Comment:
Since we're still sending an actual `count`, why the rename?

Reply:
left over from previous test using gauge